src/api/interfaces/ApiResult.ts

```diff
@@ -0,0 +1,11 @@
+// standardized return format for api result
+interface ApiResult {
+	success: boolean;
+	message: string;
+	errors?: string[];
+	data: object | unknown[] | null;
```

Comment:
Question:

Can we just make the `data` optional by declaring it as `data?`. This way we do not need to type it as potentially `null`? I think having a `null` is required for GraphQL but I do not know if it is required for REST API based. But I can be wrong here so please correct me.

Reply:
Good point, will make it optional!